code/game/objects/structures/crates_lockers/closets/secure/supplypod.dm

```diff
@@ -47,17 +56,18 @@
 /obj/structure/closet/supplypod/open()
 	var/turf/T = get_turf(src)
 	opened = TRUE
-	if(!generated)
-		generated = TRUE
+	if(SupplyOrder)
 		SupplyOrder.generate(T)//not called during populateContents as supplyorder generation requires a turf
+	if(other_delivery)
+		new other_delivery(T)
 	update_icon()
 	playsound(src, open_sound, 15, 1, -3)
 	if(istype(src,/obj/structure/closet/supplypod/bluespacepod))
-		addtimer(CALLBACK(src, .proc/sparks), 30)//if bluespace, then 3 seconds after opening, make some sparks and delete
-
-/obj/structure/closet/supplypod/proc/sparks()//sparks cant be called from addtimer
- 	do_sparks(5, TRUE, src)
- 	qdel(src)//no need for QDEL_IN if we already have a timer
+		addtimer(CALLBACK(src, .proc/sparks), 30)//if bluespace, then 3 seconds after opening, make some sparks and delete		
+ 		
+/obj/structure/closet/supplypod/proc/sparks()//sparks cant be called from addtimer		
+ 	do_sparks(5, TRUE, src)		
+ 	qdel(src)//no need for QDEL_IN if we already have a timer 

```

Comment:
Can you undo these spurious whitespace changes?

Reply:
whup i retard